src/parse.jl

```diff
@@ -0,0 +1,254 @@
+                if field["type"] == "real"
+                    id_variable += 1
+                    add_variable!(HDict["variables"], field, id_variable)
+                elseif field["type"] == "label"
+                    add_transition_label!(HDict["transitionlabels"], field)
+                else
+                    error("param type unknown")
+                end
+            elseif nodename(field) == "location"
+                (id_location, i, f) = parse_location(field)
+                HDict["invariants"][id_location] = i
+                HDict["flows"][id_location] = f
+            elseif nodename(field) == "transition"
+                id_transition += 1
+                (q, r, guard, assignment) = parse_transition(field)
+                add_transition!(HDict["automaton"], q, r, σ)
+                HDict["switchings"][id_transition] = guard
+                HDict["resetmaps"][id_transition] = assignment
+            else
+                error("node $(nodename(field)) unknown")
+            end
+        end
+    end
+    return HDict
+end
+
+"""
+    add_variable!(variables, field, id=1)
+
+### Input
+
+- `variables` -- vector of symbolic variables
+- `field`     -- node with a `param` variable field
+- `id`        -- (optional, default: 1) integer that identifies the variable
+
+### Output
+
+The updated vector of symbolic variables.
+
+### Notes
+
+Parameters can be either variable names (type "real") or labels (type "label").
+"""
+function add_variable!(variables, field, id=1)
+    @assert field["type"] == "real"
+
+    # d1 and d2 are the dimensions (d1=d2=1 for scalars)
+    @assert field["d1"] == "1" && field["d2"] == "1"
+
+    varname = parse(field["name"])
+    islocal = haskey(field, "local") ? parse(field["local"]) : false
+    iscontrolled = haskey(field, "controlled") ? parse(field["controlled"]) : true
+    dynamicstype = haskey(field, "dynamics") ? field["dynamics"] : "any"
+
+    variables[varname] = Dict("local"=>islocal,
+                              "controlled"=>iscontrolled,
+                              "dynamics"=>dynamicstype,
+                              "id"=>id)
+    return variables
+end
+
+"""
+    add_transition_label!(labels, field)
+
+### Input
+
+- `labels` -- vector of transition labels
+- `field`  -- node with a `param` label field
+
+### Output
+
+The updated vector of transition labels.
+
+### Notes
+
+Parameters can be either variable names (type "real") or labels (type "label").
+"""
+function add_transition_label!(labels, field)
+    @assert field["type"] == "label"
+    push!(labels, field["name"])
+    return labels
+end
+
+"""
+    parse_location(field)
+
+### Input
+
+- `field`  -- location node
+
+### Output
+
+The tuple `(id, I, f)` where `id` is the integer that identifies the location,
+`I` is the list of invariants for this location, and `f` is the list of ODEs that
+define the flow. Both objects are vectors of symbolic expressions `Expr`.
```

Comment:
why lists? There is only one invariant/flow per location. You mean "list of subexpressions", right?

Reply:
yes, list of subexpressions is more precise, proposal:

```julia
### Output

The tuple `(id, I, f)` where `id` is the integer that identifies the location,
`I` is the list of subexpressions that determine that invariant for this location,
and similarly `f` is the list of ODEs that define the flow for this location.
Both objects are vectors of symbolic expressions `Expr`.
```